### PR TITLE
Minor clarifications around Readable._read and Readable.push

### PR DESCRIPTION
**Pull request by fresheneesz**
https://github.com/joyent/node/issues/14124#issuecomment-115392674

### CODE REVIEW

File: doc/api/stream.markdown [hunk 1]
```diff
@@ -963,19 +961,16 @@ becomes available.  There is no need, for example to "wait" until
...
+Note: **This method should be called by Readable implementors, NOT
 by consumers of Readable streams.**
 
-The `_read()` function will not be called again until at least one
-`push(chunk)` call is made.
-
-The `Readable` class works by putting data into a read queue to be
-pulled out later by calling the `read()` method when the `'readable'`
-event fires.
+If a String or Buffer is passed, The `push()` method adds a chunk of data
```
Comment (chrisdickinson) [inline doc/api/stream.markdown]: This gets a little into the weeds, but this should be "if a non-null value is passed, the `push()` method adds it into the queue for subsequent..." (taking into account `objectMode`.)

**Comment by fresheneesz** [comment]
Changed the wording as per chris dickinson's comment

**Comment by jasnell** [comment]
Looks like a good change overall. Would appreciate if you could squash the commits down into a single commit and add a bit more of a summary to the commit message (helpful but not critical). Thank you!

**Comment by fresheneesz** [comment]
@jasnell Is there an easy way to do that in github or do I have to pull the whole repo?

**Comment by fresheneesz** [comment]
Ugh, what an unintuitive process. Ok I squashed the commits into this one: https://github.com/fresheneesz/node/commit/0a09f303c299794f6ab13c19a97b310783723e39 . What do I do now? Close this one and create a new pull request?

**Comment by jasnell** [comment]
Nope! Nothing further is necessary on your part. I'll take another pass at reviewing this shortly then will work on getting it landed (likely next week). Thank you!

**Comment by fresheneesz** [comment]
Ok! Sure thing!

**Comment by jasnell** [comment]
LGTM. @chrisdickinson @trevnorris

**Comment by trevnorris** [comment]
LGTM

**Comment by jasnell** [comment]
@fresheneesz ... unfortunately, this appears to have been clobbered a bit by another change that was landed. If you'd like to rebase and resolve the conflict, that would be great. I'll get it landed right after. Otherwise, once I'm through a few other items I'll see if I can get to rebasing this.

**Comment by fresheneesz** [comment]
: / , I don't have time right now to figure that out, but maybe in a couple weeks.

**Comment by jasnell** [comment]
No worries, if I get the time before then I'll untangle it and point here with the new commit.

**Comment by jasnell** [comment]
Ok landed this in 4952e2b and over in io.js as well.

**Pull request closed without merging**